app/uk/gov/hmrc/gform/upscan/UpscanController.scala

```diff
@@ -140,22 +142,25 @@ class UpscanController(
                   case _                                                 => false
                 }
                 for {
-                  file <- upscanService.download(upscanCallbackSuccess.downloadUrl)
-                  compressedFile = ImageCompressor.compressIfSupported(file, upscanCallbackSuccess, compression)
-                  uploadDetails = upscanCallbackSuccess.uploadDetails
                   isObjectStore <- objectStoreAlgebra.isObjectStore(envelopeId)
+                  uploadDetails = upscanCallbackSuccess.uploadDetails
                   _ <- if (isObjectStore) {
-
-                         objectStoreAlgebra.uploadFile(
+                         objectStoreAlgebra.uploadFromUrl(
+                           new URL(upscanCallbackSuccess.downloadUrl),
                            envelopeId,
                            fileId,
-                           fileName,
-                           compressedFile,
-                           ContentType(uploadDetails.fileMimeType)
+                           ContentType(uploadDetails.fileMimeType),
+                           fileName
                          )
                        } else {
-                         fileUploadFrontendAlgebra
-                           .uploadFile(envelopeId, fileId, uploadDetails, compressedFile)
+                         for {
+                           file <- upscanService.download(upscanCallbackSuccess.downloadUrl)
+                           compressedFile =
+                             ImageCompressor.compressIfSupported(file, upscanCallbackSuccess, compression)
+                           res <- fileUploadFrontendAlgebra
+                                    .uploadFile(envelopeId, fileId, uploadDetails, compressedFile)
```

Comment:
Doing the compression is sort of reverting the advantage of using `uploadFromUrl` so we should drop doing it.

Reply:
I would like to keep this compression if the form does not use the object store.

Reply:
Removed